docs/model_access_control.md

```diff
@@ -0,0 +1,652 @@
+* For a private model group of User1, only User1 or admin will be able to create new versions. For any other user, above exception will be thrown.
+* For a public model group of User1, any user can access register a version.
+
+
+#### Example Request: Registering a model version via URL
+
+```
+POST /_plugins/_ml/models/_register
+
+{
+    "name": "all-MiniLM-L6-v2",
+    "version": "1.0.0",
+    "description": "test model",
+    "model_format": "TORCH_SCRIPT",
+    "model_group_id": "FTNlQ4gBYW0Qyy5ZoxfR",
+    "model_content_hash_value": "9376c2ebd7c83f99ec2526323786c348d2382e6d86576f750c89ea544d6bbb14",
+    "model_config": {
+        "model_type": "bert",
+        "embedding_dimension": 384,
+        "framework_type": "sentence_transformers",
+       "all_config": "{\"_name_or_path\":\"nreimers/MiniLM-L6-H384-uncased\",\"architectures\":[\"BertModel\"],\"attention_probs_dropout_prob\":0.1,\"gradient_checkpointing\":false,\"hidden_act\":\"gelu\",\"hidden_dropout_prob\":0.1,\"hidden_size\":384,\"initializer_range\":0.02,\"intermediate_size\":1536,\"layer_norm_eps\":1e-12,\"max_position_embeddings\":512,\"model_type\":\"bert\",\"num_attention_heads\":12,\"num_hidden_layers\":6,\"pad_token_id\":0,\"position_embedding_type\":\"absolute\",\"transformers_version\":\"4.8.2\",\"type_vocab_size\":2,\"use_cache\":true,\"vocab_size\":30522}"
+    },
+    "url": "https://github.com/ylwu-amzn/ml-commons/blob/2.x_custom_m_helper/ml-algorithms/src/test/resources/org/opensearch/ml/engine/algorithms/text_embedding/all-MiniLM-L6-v2_torchscript_sentence-transformer.zip?raw=true"
+}
+```
+
+```
+#Response
+{
+    "task_id": "FjNlQ4gBYW0Qyy5Zxhdh",
+    "status": "CREATED"
+}
+```
+
+
+
+## Deploy/Undeploy/Predict/Delete/Get model version
+
+On a model_access_control enabled cluster:
+* only owner or users who have atleast one of their backend roles matching with that of a model version can perform deploy/undeploy/predict/delete/get actions on it.
+* users can also perform the said actions on any version that belongs to a public model group.
+* For a private model version, only its owner or the admin can perform these actions.
+* Admin can perform all these actions on any model version in the cluster
+
+On a security/model_access_control disabled cluster,
+* a user can perform the actions to which they have permissions to on any model version.
+
+For example:
+* For model versions registered in a model group created by User1 with IT backend_role, only user1, user2 and admin can perform the above actions. If user3 and user4 try to send a request, an exception will be thrown: `You don't have permissions to perform this operation on this model.`
+* For model versions registered in a private model group of User1, only User1 or admin will be able to perform the above actions on its versions. For any other user, above exception will be thrown.
+* For a public model group of User1, any user can perform the said actions its versions.
+
+
+
+
+## Search model versions
```

Comment:
Add some example search request? Like how to search all model versions in a model group with model group id